drivers/include/periph/rtc.h

```diff
@@ -74,9 +76,11 @@ int rtc_get_time(struct tm *time);
  * @param[in] cb            Callback executed when alarm is hit.
  * @param[in] arg           Argument passed to callback when alarm is hit.
  *
- * @return  0 for success
- * @return -2 invalid `time` parameter
- * @return -1 other errors
+ * @retval  0               Success
+ * @retval -EINVAL          @p time or @p cb is `NULL`
+ * @retval -EINVAL          @p time does not contain a valid time
+ * @retval -ETIME           @p time refers to the past (alarm would never trigger)
```

Comment:
No error code fully matches this. These would be the closes matches:

| Error Code | Output of `strerror(errno)`                    |
|:---------- |:---------------------------------------------- |
| `ETIME`    | Timer expired                                  |
| `EINVAL`   | Invalid argument                               |
| `EDOM`     | Mathematics argument out of domain of function |

The best match would be `-EINVAL` for timers in the past. But I think there is value if the caller is able to distinguish the causes for the fault.

Reply:
I would prefer `-ETIME`.